src/main/java/net/sf/jabref/logic/net/URLDownload.java

```diff
@@ -186,17 +188,55 @@ private void copy(InputStream in, Writer out, Charset encoding) throws IOExcepti
         }
     }
 
+    /**
+     * @deprecated use {@link #downloadToFile(Path)}
+     */
+    @Deprecated
     public void downloadToFile(File destination) throws IOException {
+        downloadToFile(destination.toPath());
+    }
+
+    public void downloadToFile(Path destination) throws IOException {
 
         try (InputStream input = new BufferedInputStream(openConnection().getInputStream());
-             OutputStream output = new BufferedOutputStream(new FileOutputStream(destination))) {
+                OutputStream output = new BufferedOutputStream(new FileOutputStream(destination.toFile()))) {
             copy(input, output);
         } catch (IOException e) {
             LOGGER.warn("Could not copy input", e);
             throw e;
         }
     }
 
+    /**
+     * Downloads the web resource to a temporary file.
+     *
+     * @return the path to the downloaded file.
+     */
+    public Path downloadToTemporaryFile() throws IOException {
+        // Determine file name and extension from source url
+        String sourcePath = source.getPath();
+
+        // Take everything after the last '/' as name + extension
```

Comment:
`source` is a URL and thus `source.getPath()` is just a string and not a nio.Path